code/modules/sanity/sanity_mob.dm

```diff
@@ -95,18 +95,15 @@
 	if(sanity_invulnerability)
 		return
 	var/vig = owner.stats.getStat(STAT_VIG)
-	var/sanity_damage_view = 0
 	for(var/atom/A in view(owner.client ? owner.client : owner))
```

Comment:
not quite a gripe towards your changes, but why does it iterate through atoms in view instead of iterating through humanmobs in the view?